### PR TITLE
[FIX] "*.members" rest api being useless and only returning usernames

### LINKED ISSUE #8116: [API] Improve members list API, adding name, status and userId to the list
### Description:

The `members` APIs are only returning a list of usernames, witch is not useful. We need to have more information about the user: userId, name and status at least.

### PR DESCRIPTION
**Pull request by graywolf336**
@RocketChat/core 

Closes #8116

### CODE REVIEW

**Comment by sampaiodiego** [comment]
am I wrong or I could exploit this to get any user's field?

**Comment by graywolf336** [comment]
You're not wrong :-1:

We almost need a list of fields in the system which a normal user can access and which ones an admin can access. I can change things so that only the `_id, username, status` are accessible via this...

**Comment by sampaiodiego** [comment]
I would say to change it so only `_id, username, name, status, utcOffset` are returned to regular users..

admins could request any additional field.

**Comment by filipedelimabrito** [comment]
Any chance to return the user avatar on this rest api call? Because we always show the avatar when we need to see the member list of a room.

**Comment by graywolf336** [comment]
@filipedelimabrito the `avatar` property on the messages isn't always promised to be there as that's a special customization property, so no that won't be returned unless the message is special and has it.

**Comment by filipedelimabrito** [comment]
Thanks, but how can I get the user avatar?

**Comment by graywolf336** [comment]
@filipedelimabrito `$ROOTURL/avatar/$USERNAME`

For example: https://demo.rocket.chat/avatar/bradley.hilton

File: packages/rocketchat-api/server/v1/channels.js [hunk 1]
```diff
@@ -422,16 +422,19 @@ RocketChat.API.v1.addRoute('channels.members', { authRequired: true }, {
...
+		const { sort, fields } = this.parseJsonQuery();
 
 		const members = RocketChat.models.Rooms.processQueryOptionsOnResult(Array.from(findResult.usernames), {
 			sort: sort ? sort : -1,
 			skip: offset,
 			limit: count
 		});
 
+		const ourFields = Object.assign({ _id: 1, username: 1, status: 1 }, fields, RocketChat.API.v1.defaultFieldsToExclude);
+		const users = RocketChat.models.Users.find({ username: { $in: members } }, { fields: ourFields }).fetch();
```
Comment (rodrigok) [inline packages/rocketchat-api/server/v1/channels.js]: How heavy is this find on large rooms?
What about use cache for that?
```
RocketChat.models.Subscriptions.find({rid: 'GENERAL'}).forEach(item => console.log(item._user))
```

Comment (graywolf336) [reply packages/rocketchat-api/server/v1/channels.js]: It's limited to 50 by default but users can request more and if they request a ton, then that load is on them.

**Pull request merged**